From 0f449aea1113949faed887c391427103c896a8dd Mon Sep 17 00:00:00 2001
From: Priyajit Ghosh <prghosh@habana.ai>
Date: Mon, 17 Mar 2025 06:04:30 +0200
Subject: [PATCH] [FSW-14191] fs1: enable force recovery in coldboot

In this patch we enable force recovery during cold
boot. If force recovery is enable it will go to
recovery insted of manifest0, manifest1 is valid.

Change-Id: I4af2e9dc9f7689e61d8e71e3585591b147c7759c
Signed-off-by: Priyajit Ghosh <prghosh@habana.ai>
---

diff --git a/app/boot_rom.c b/app/boot_rom.c
index 37632bf..13ebfe0 100644
--- a/app/boot_rom.c
+++ b/app/boot_rom.c
@@ -65,9 +65,12 @@
 	/* Wait for ppboot spi dma to complete */
 	boot_rom_boot_flow_wait_for_dma_finish();
 
-	/* If both manifests are invalid enter into recovery */
-	if (!boot_rom_boot_flow_is_manifest_valid(BOOT_ROM_PRIMARY) &&
-		!boot_rom_boot_flow_is_manifest_valid(BOOT_ROM_SECONDARY)) {
+	/* If both manifests are invalid or force recovery is set then enter
+	 * into recovery
+	 */
+	if ((!boot_rom_boot_flow_is_manifest_valid(BOOT_ROM_PRIMARY) &&
+		!boot_rom_boot_flow_is_manifest_valid(BOOT_ROM_SECONDARY)) ||
+		(boot_rom_global.force_recovery == true)) {
 		execute_recovery();
 	}
 
